### PR TITLE
Run / Debug configurations and separated server for development

### PR DESCRIPTION
**Pull request by sascha11110**
### Motivation and Context
Are you tired that the client is often built, even if you just want to test your changes to the server files? You are using the separated client via `yarn start` anyway and you hate that there is an additional node process bloating up your CPU? Then this is for you! You can now start the server and the client seperately via different run configurations.

Since the run / debug configurations had to be adjusted here anyway, I have added more profiles to make the setup easier.

### Description
- [x] I’ve added a couple of default run / debug configurations that will be loaded by IntelliJ automatically (thanks to @sleiss for the help):
  - **Artemis (Server)**: The server will be started separated from the client. The startup time decreases significantly.
   - **Artemis (Client)**: Will execute `yarn install` and `yarn start`. The client will be available at http://localhost:9000/ with hot reload enabled.
  - **Artemis (Server & Client)**: Will start the server and the client. The client will be available at http://localhost:8080/ with hot reload disabled.
  - **Artemis (Server, Jenkins & Gitlab)**: The server will be started separated from the client with the profiles `dev,jenkins,gitlab,artemis` instead of `dev,bamboo,bitbucket,jira,artemis`.
  - **Artemis (Server, Text Clustering)**: The server will be started separated from the client with `automaticText` profile enabled (see https://github.com/ls1intum/Artemis/blob/develop/docs/dev/setup.rst#text-assessment-clustering-service).
  - **Artemis JavaScript Debug**: The configuration for the JavaScript debugger. Will use the system’s default browser.
- [x] I’ve updated the documentation for the setup of the local development setup.
- [x] I've restored the .gitignore configuration to make sure that local changes are not accidentality pushed.

All configurations set the Spring profiles via `-Dspring.profiles.active=dev,bamboo,bitbucket,jira,artemis,automaticText` in the `vm parameter` options as setting the Spring profiles directly is limited to IntelliJ Ultimate.

### Steps for Testing
Checkout this branch locally, restart IntelliJ and test all run / debug configurations. Especially the profiles for *Artemis (Jenkins & Gitlab)* and *Artemis (Server, Text Clustering)* if you have worked with it before.

### Screenshots
image

### CODE REVIEW

File: docs/dev/setup.rst [hunk 1]
```diff
@@ -159,12 +181,15 @@ command in your terminal / command line:
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The Artemis server should startup by running the main class
+``de.tum.in.www1.artemis.ArtemisApp`` using Spring Boot.
 
-.. note:: 
+.. note::
     Artemis uses Spring profiles to segregate parts of the
-    application configuration and make it only available in certain
+    application configuration and mae it only available in certain
```
Comment (filip-gregurevic) [inline docs/dev/setup.rst:192]:
```suggestion
    application configuration and make it only available in certain
```
small typo

File: docs/dev/setup.rst [hunk 2]
```diff
@@ -226,7 +274,7 @@ file and save, the client is automatically reloaded with the new code)
 and will start the client application in your browser on
 ``http://localhost:9000``. If you have activated the JIRA profile (see
 above in Server Setup) and if you have configured
-``application-artemis.yml`` correctly, then you should be able to login
+``application-artemis.yml`` correctly, then you should be able to loginâ
```
Comment (filip-gregurevic) [inline docs/dev/setup.rst:277]:
```suggestion
``application-artemis.yml`` correctly, then you should be able to login
```
don't know what that is

**Comment by sascha11110** [comment]
@tessaRuckstuhl @sleiss @filip-gregurevic I had to fix some minor format issues in the documentation. Would be nice if you could re-approve. Testing the configurations is not necessary as I haven't changed something there.

File: .idea/runConfigurations/Artemis__Client_.xml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<component name="ProjectRunConfigurationManager">
+  <configuration default="false" name="Artemis (Client)" type="js.build_tools.npm">
+    <package-json value="$PROJECT_DIR$/package.json" />
+    <command value="run" />
+    <scripts>
+      <script value="start" />
```
Comment (jpbernius) [inline .idea/runConfigurations/Artemis__Client_.xml:6]:
```suggestion
      <script value="webpack:dev" />
```

File: .idea/runConfigurations/Artemis__Server_.xml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<component name="ProjectRunConfigurationManager">
+  <configuration default="false" name="Artemis (Server)" type="SpringBootApplicationConfigurationType" factoryName="Spring Boot">
+    <module name="Artemis.main" />
+    <option name="SPRING_BOOT_MAIN_CLASS" value="de.tum.in.www1.artemis.ArtemisApp" />
+    <option name="VM_PARAMETERS" value="-Dspring.profiles.active=dev,bamboo,bitbucket,jira,artemis" />
```
Comment (jpbernius) [inline .idea/runConfigurations/Artemis__Server_.xml:5]:
```suggestion
    <option name="ACTIVE_PROFILES" value="dev,bamboo,bitbucket,jira,artemis" />
```

File: .idea/runConfigurations/Artemis__Server__Jenkins___Gitlab_.xml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<component name="ProjectRunConfigurationManager">
+  <configuration default="false" name="Artemis (Server, Jenkins &amp; Gitlab)" type="SpringBootApplicationConfigurationType" factoryName="Spring Boot">
+    <module name="Artemis.main" />
+    <option name="SPRING_BOOT_MAIN_CLASS" value="de.tum.in.www1.artemis.ArtemisApp" />
+    <option name="VM_PARAMETERS" value="-Dspring.profiles.active=dev,jenkins,gitlab,artemis" />
```
Comment (jpbernius) [inline .idea/runConfigurations/Artemis__Server__Jenkins___Gitlab_.xml:5]:
```suggestion
    <option name="ACTIVE_PROFILES" value="dev,jenkins,gitlab,artemis" />
```

File: .idea/runConfigurations/Artemis__Server__Text_Clustering_.xml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<component name="ProjectRunConfigurationManager">
+  <configuration default="false" name="Artemis (Server, Text Clustering)" type="SpringBootApplicationConfigurationType" factoryName="Spring Boot">
+    <module name="Artemis.main" />
+    <option name="SPRING_BOOT_MAIN_CLASS" value="de.tum.in.www1.artemis.ArtemisApp" />
+    <option name="VM_PARAMETERS" value="-Dspring.profiles.active=dev,bamboo,bitbucket,jira,artemis,automaticText" />
```
Comment (jpbernius) [inline .idea/runConfigurations/Artemis__Server__Text_Clustering_.xml:5]:
```suggestion
    <option name="ACTIVE_PROFILES" value="dev,bamboo,bitbucket,jira,artemis,automaticText" />
```

File: .idea/runConfigurations/Artemis__Server___Client_.xml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+<component name="ProjectRunConfigurationManager">
+  <configuration default="false" name="Artemis (Server &amp; Client)" type="SpringBootApplicationConfigurationType" factoryName="Spring Boot">
+    <module name="Artemis.main" />
+    <option name="SPRING_BOOT_MAIN_CLASS" value="de.tum.in.www1.artemis.ArtemisApp" />
+    <option name="VM_PARAMETERS" value="-Dspring.profiles.active=dev,bamboo,bitbucket,jira,artemis" />
```
Comment (jpbernius) [inline .idea/runConfigurations/Artemis__Server___Client_.xml:5]:
```suggestion
    <option name="ACTIVE_PROFILES" value="dev,bamboo,bitbucket,jira,artemis" />
```

File: docs/dev/setup.rst [hunk 8]
```diff
@@ -144,8 +144,30 @@ information about the setup for programming exercises provided:
...
+
+* **Artemis (Server):** The server will be started separated from the client. The startup time decreases significantly.
+* **Artemis (Client):** Will execute ``yarn install`` and ``yarn start``. The client will be available at `http://localhost:9000/ <http://localhost:9000/>`__ with hot module replacement enabled (also see `Client Setup <https://github.com/ls1intum/Artemis/blob/develop/docs/dev/setup.rst#client-setup>`__).
+
+Other run / debug configurations
+""""""""""""""""""""""""""""""""
+
+* **Artemis (Server & Client):** Will start the server and the client. The client will be available at `http://localhost:8080/ <http://localhost:8080/>`__ with hot module replacement disabled.
+* **Artemis (Server, Jenkins & Gitlab):** The server will be started separated from the client with the profiles ``dev,jenkins,gitlab,artemis`` instead of ``dev,bamboo,bitbucket,jira,artemis``.
+* **Artemis (Server, Text Clustering):** The server will be started separated from the client with ``automaticText`` profile enabled (see `Text Assessment Clustering Service <https://github.com/ls1intum/Artemis/blob/develop/docs/dev/setup.rst#text-assessment-clustering-service>`__).
```
Comment (jpbernius) [inline docs/dev/setup.rst:166]:
```suggestion
* **Artemis (Server, Text Clustering):** The server will be started separated from the client with ``automaticText`` profile enabled (see `Text Assessment Clustering Service <setup.rst#text-assessment-clustering-service>`__).
```

File: docs/dev/setup.rst [hunk 9]
```diff
@@ -144,8 +144,30 @@ information about the setup for programming exercises provided:
...
+Run the server via a run configuration in IntelliJ
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The project comes with some pre-configured run / debug configurations that are stored in the ``.idea`` directory.
+When you import the project into IntelliJ the run configurations will also be imported.
+
+The recommended way is to run the server and the client separated. This provides fast rebuilds of the server and hot module replacement in the client.
+
+* **Artemis (Server):** The server will be started separated from the client. The startup time decreases significantly.
+* **Artemis (Client):** Will execute ``yarn install`` and ``yarn start``. The client will be available at `http://localhost:9000/ <http://localhost:9000/>`__ with hot module replacement enabled (also see `Client Setup <https://github.com/ls1intum/Artemis/blob/develop/docs/dev/setup.rst#client-setup>`__).
```
Comment (jpbernius) [inline docs/dev/setup.rst:159]:
```suggestion
* **Artemis (Client):** Will execute ``yarn install`` and ``yarn start``. The client will be available at `http://localhost:9000/ <http://localhost:9000/>`__ with hot module replacement enabled (also see `Client Setup <setup.rst#client-setup>`__).
```

File: .idea/runConfigurations/Artemis__Server_.xml (hunk 4 above)
Comment (sascha11110) [reply .idea/runConfigurations/Artemis__Server_.xml:5]: I haven't used the `ACTIVE_PROFILES` option on purpose as the option is only available in IntelliJ Ultimate and some of the developers only use the Community Edition. Therefore I will not add your suggestion for this and the other files :-)

**Comment by sascha11110** [comment]
@jpbernius I haven't used the ACTIVE_PROFILES option on purpose as the option is only available in IntelliJ Ultimate and some of the developers only use the Community Edition. Therefore I only added the other suggestions :-)

**Pull request merged**